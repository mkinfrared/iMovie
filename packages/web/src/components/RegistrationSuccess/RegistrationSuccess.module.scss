.RegistrationSuccess {
  margin: 0 auto;
  max-width: 480px;
  text-align: center;

  svg {
    width: 75%;
  }
}
