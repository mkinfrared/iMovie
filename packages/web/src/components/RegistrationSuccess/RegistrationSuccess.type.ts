export interface RegistrationSuccessProps {
  onClose: () => void;
  open: boolean;
}
