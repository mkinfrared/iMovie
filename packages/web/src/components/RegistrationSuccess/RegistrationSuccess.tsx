import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import React from "react";

import Email from "icons/Email";
import Transition from "ui/Transition";

import css from "./RegistrationSuccess.module.scss";
import { RegistrationSuccessProps } from "./RegistrationSuccess.type";

const RegistrationSuccess = ({ onClose, open }: RegistrationSuccessProps) => {
  return (
    <Dialog
      open={open}
      className={css.RegistrationSuccess}
      TransitionComponent={Transition}
      onClose={onClose}
    >
      <DialogContent>
        <DialogContentText>
          <Email />
        </DialogContentText>
        <DialogContentText>
          Check your email to proceed with registration
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose} color="primary" data-testid="registerButton">
          OK
        </Button>
      </DialogActions>
    </Dialog>
  );
};

export default React.memo(RegistrationSuccess);
