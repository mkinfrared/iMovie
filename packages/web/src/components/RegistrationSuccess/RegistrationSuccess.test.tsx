import { render } from "@testing-library/react";
import React, { ReactElement } from "react";

import RegistrationSuccess from "./RegistrationSuccess";

describe("<RegistrationSuccess />", () => {
  const open = true;
  const onClose = jest.fn();

  let Component: ReactElement;

  beforeEach(() => {
    Component = <RegistrationSuccess open={open} onClose={onClose} />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
