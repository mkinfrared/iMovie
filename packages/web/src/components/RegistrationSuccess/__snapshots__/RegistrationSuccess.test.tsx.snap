// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationSuccess /> should match a snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
